### PR TITLE
Remove unused script-security dependency

### PR DESCRIPTION
**Pull request by basil**
Whatever the original reasons for introducing this `test`-scoped dependency were, they do not seem to be valid any longer. Nothing in this source tree consumes the `org.jenkinsci.plugins.scriptsecurity` package.

### CODE REVIEW

**Review by oleg-nenashev — APPROVED** [review]
+1 since it is unused, but in general we should ensure that there is some test coverage for Script Security and Pipeline in the Jenkins core builds. It is a key feature, and maybe it is not a huge deal to integrate if we bring Plugin BOM into this test suite

**Comment by daniel-beck** [comment]
> Whatever the original reasons

https://github.com/jenkinsci/jenkins/blob/4b90b9ab16b0be0fbfa6f7ad2b319c648c8aa472/test/pom.xml#L129

**Review by timja — APPROVED** [review]

**Comment by timja** [comment]
This PR is now ready for merge, after ~24 hours, we will merge it if there's no negative feedback.

Thanks!

**Pull request merged**
